atc/db/wall.go

```diff
@@ -24,62 +23,87 @@ func NewWall(conn Conn) Wall {
 	return &wall{conn: conn}
 }
 
-func (w wall) SetMessage(message string) error {
-	err := w.Clear()
+func (w wall) SetWall(wall atc.Wall) error {
+	tx, err := w.conn.Begin()
 	if err != nil {
 		return err
 	}
 
-	_, err = psql.Insert("wall").
-		Columns("message").
-		Values(message).
-		RunWith(w.conn).Exec()
+	defer Rollback(tx)
+
+	_, err = psql.Delete("wall").RunWith(tx).Exec()
+	if err != nil {
+		return err
+	}
+
+	query := psql.Insert("wall").
+		Columns("message")
+
+	if wall.TTL != "" {
+		ttl, err := time.ParseDuration(wall.TTL)
+		if err != nil {
+			return err
+		}
```

Comment:
It's hard to feel ethical rolling back a transaction because I failed to deserialize some data - feels a bit like cancelling a meeting because I have to go to the bathroom.

I think that the `TTL` field on `atc.Wall` should be a `time.Duration` instead of a string and this logic should get moved into the API layer, since it's really a serialization/validation issue.

Reply:
the bathroom analogy won me over 😆